Docs build error in Sphinx 4.0.0

## 🐛 Bug
I locally confirmed that building docs (`cd docs; make html`) fails due to:
```
Extension error (sphinx_paramlinks.sphinx_paramlinks):
Handler <function build_index at 0x7f20c201c820> for event 'doctree-read' threw an exception (exception: __new__() missing 1 required positional argument: 'aliased')
make: *** [Makefile:19: html] Error 2
Traceback (most recent call last):
  File "/entrypoint.py", line 22, in <module>
    action.build_all_docs(github_env, [os.environ.get("INPUT_DOCS-FOLDER")])
  File "/sphinx_action/action.py", line 167, in build_all_docs
    raise RuntimeError("Build failed")
RuntimeError: Build failed
```
and also confirmed that `pip install Sphinx<4.0.0` fixes the error.

### To Reproduce
https://github.com/PyTorchLightning/pytorch-lightning/pull/7452/checks?check_run_id=2538915517

### Expected behavior
`make html` should successfully build the docs.

--- a/requirements/docs.txt
+++ b/requirements/docs.txt
@@ -1,4 +1,4 @@
-sphinx>=3.0, !=3.5, <4.0  # fails with sphinx.ext.viewcode  # fails with sphinx.paramlinks with 4.0.0
+sphinx>=3.0, !=3.5, <4.0  # fails with sphinx.ext.viewcode  # fails with sphinx_paramlinks with 4.0.0
 recommonmark  # fails with badges
 m2r  # fails with multi-line text
 nbsphinx>=0.8

--- a/requirements/docs.txt
+++ b/requirements/docs.txt
@@ -1,4 +1,4 @@
-sphinx>=3.0, !=3.5, <4.0  # fails with sphinx.ext.viewcode  # fails with sphinx_paramlinks with 4.0.0
+sphinx>=3.0, <3.5  # fails with sphinx.ext.viewcode  # fails with sphinx_paramlinks with 4.0.0
 recommonmark  # fails with badges
 m2r  # fails with multi-line text
 nbsphinx>=0.8

--- a/requirements/docs.txt
+++ b/requirements/docs.txt
@@ -4,6 +4,7 @@ m2r  # fails with multi-line text
 nbsphinx>=0.8
 pandoc>=1.0
 docutils>=0.16
+Sphinx<4.0.0
 sphinxcontrib-fulltoc>=1.0
 sphinxcontrib-mockautodoc
 https://github.com/PyTorchLightning/lightning_sphinx_theme/archive/master.zip#egg=pt-lightning-sphinx-theme

--- a/requirements/docs.txt
+++ b/requirements/docs.txt
@@ -1,10 +1,9 @@
-sphinx>=3.0, !=3.5  # fails with sphinx.ext.viewcode
+sphinx>=3.0, !=3.5, <4.0  # fails with sphinx.ext.viewcode  # fails with sphinx.paramlinks with 4.0.0
 recommonmark  # fails with badges
 m2r  # fails with multi-line text
 nbsphinx>=0.8
 pandoc>=1.0
 docutils>=0.16
-Sphinx<4.0.0
 sphinxcontrib-fulltoc>=1.0
 sphinxcontrib-mockautodoc
 https://github.com/PyTorchLightning/lightning_sphinx_theme/archive/master.zip#egg=pt-lightning-sphinx-theme

